### ISSUE TITLE
Section custom template

### ISSUE THREAD

**Issue by bactisme** [themosis/framework#224]
The Section::make constructor seems to accept a view. But template doesn't show on the page.

This doesn't work: 

    add_action('init', function () {

        $newsletter_page = Page::make('newsletter-hebdo', 'Newsletter Hebdo', null, $admin_view)->set(array(
            'capability' => 'manage_options',
            'icon' => 'dashicons-admin-site',
            'tabs' => true
        ));

        $admin_view = View::make('admin.send-hebdo');
        $newsletter_page->addSections([
            Section::make('newsletter-hebdo-posts', 'Newsletter Hebdo'),
            Section::make('newsletter-hebdo-send', 'Envoyer', [], $admin_view)
         ]); 
    (...)

**Comment by jlambe** [comment]
Yes but it's not used currently in the framework. If you look at PageBuilder class, displaySections method:
https://github.com/themosis/framework/blob/master/src/Themosis/Page/PageBuilder.php#L361

The method is currently empty so it only displays the section name.

This feature is planned but in the meantime I can give you tips to add this feature: the `$args` parameter contains the section `id`, `title`, `callback` keys. The `$args['callback'][0]` is the page instance object.

We have to put some logic in order to filter registered sections within the page, grab its view, render it and echo it.

I'll point this to release 1.3.0. Note that the view for your section will be displayed under your section title tag which I don't know if we can overwrite it.

**Comment by bactisme** [comment]
By the way. It could be super cool to have a do_action when saving settings. Could be useful for many things like flushing cache etc.

**Comment by jlambe** [comment]
Well I won't add those hooks as they are already defined inside the WordPress core. Normally the settings API use the `add/update/delete_option()` functions which all have action hooks to use before and after the data has been saved.

##### add_option
Before being saved to DB:
`do_action("add_option", $option, $value);`

Specific option:
`do_action("add_option_{$option}", $option, $value );`

Any added option:
`do_action("added_option", $option, $value );`

##### update_option
Before being updated on DB:
`do_action("update_option", $option, $old_value, $value);``

Specific option updated:
`do_action("update_option_{$option}", $old_value, $value);`

Any updated option:
`do_action("updated_option", $option, $old_value, $value );`

I suspect you need to define both add and update hooks and specify the same callback to make sure you empty your cache.

Hope this helps.

**Comment by jlambe** [comment]
Also, by viewing your page code, why put it inside an `init` hook? Pages are created during an `admin_init` and `admin_menu` hooks, so normally you should be safe to remove your init call.
